crates/sui-framework/src/natives/object_runtime/mod.rs

```diff
@@ -159,8 +160,13 @@ impl<'a> ObjectRuntime<'a> {
         Ok(transfer_result)
     }
 
-    pub fn emit_event(&mut self, ty: Type, tag: StructTag, event: Value) {
-        self.state.events.push((ty, tag, event))
+    pub fn emit_event(&mut self, ty: Type, tag: StructTag, event: Value) -> PartialVMResult<()> {
+        if self.state.events.len() == (MAX_NUM_EVENT_EMIT as usize) {
+            return Err(PartialVMError::new(StatusCode::UNKNOWN_STATUS /* TODO (Ade): should we add a new VM StatusCode for TOO_MANY_EVENTS? */)
```

Comment:
@tnowacki do we need a new VM error here?

Reply:
yeah I really think we want a specific error here and certainly something else that unknown

Reply:
PR for adding new Move error is [here](https://github.com/move-language/move/pull/863)